### ISSUE TITLE
Flex-layout fxFlex functionality not fully supported

### ISSUE THREAD

**Issue by lievenjanssen** [philmtd/css-fx-layout#11]
I think if we could support flex-basis aliases and some column/row layouting options we could come a long way.

Possible data attribute implementation for flex-basis aliases:
```scss
*[data-fx-flex^="grow"] {
    flex: 1 1 100%;
}

*[data-fx-flex^="initial"] {
    flex: 0 1 auto;
}

*[data-fx-flex^="auto"] {
    flex-basis: auto;
}

*[data-fx-flex^="none"] {
    flex: 0 0 auto;
}

*[data-fx-flex^="nogrow"] {
    flex: 0 1 auto;
}

*[data-fx-flex^="noshrink"] {
    flex: 1 0 auto;
}
```

Possible data attribute implementation for column/row layouting (support for 5 ,10, ..., 95):
```scss
@for $i from 1 through 20 {
    *[data-fx-flex^="#{$i * 5}"] {
        flex: 1 1 $i * 5%;
    }
}
```

**Comment by philmtd** [comment]
Thank you for this suggestion! Of course we'd need an equivalent for support with CSS classes and with the responsive API, too.

To be honest I'm not sure what the best CSS class (and attribute) naming convention would be to keep it as simple and understandable as possible and I'm open for suggestions.

I'd want to make sure your loop example would be optional and configurable as it currently is with the gap property to keep the generated code base small for everyone who does not need this feature.

Would you want to implement it and open a PR?

**Comment by lievenjanssen** [comment]
Personally I would stay as close as possible to flex-layout for attribute naming. If we implement attributes the way I suggested above you can migrate from flex-layout to css-fx-layout by doing a find & replace which would be ideal for larger codebases.

My suggestions for CSS class naming would be:
- aliases: `fx-flex-grow` (already supported), `fx-flex-none`, `fx-flex-initial`, ...
- column/row layout: `fx-flex--5`, `fx-flex--10`, ... (similar to `fx-gap--`)

For the configurable part I would suggest implementing a mixin with the following arguments: from (e.g. 1), to (e.g. 100), step (e.g. 5) we could also support another mixin that you can just give a list of values (e.g. 20, 40, 60, 80, 33).

What do you think?

Once agreed upon an implementation I'm open to implement it and open a PR.

**Comment by philmtd** [comment]
Yes that sounds good and it would work with the responsive API, too, e.g.:

* `data-fx-flex-md="none"`
*  `.fx-flex-none--md`
* `.fx-flex--10--md`

I'm looking forward to your PR :) If you add your suggested gap-mixin, I think this would make a nice addition, too!

**Comment by lievenjanssen** [comment]
While working on it I was wondering if it doesn't make more sense to move the layout-gap-classes from mixin.scss to layout-classes.scss and layout-gap-attributes from mixin.scss to layout-attributes.scsss? What do you think?

I would also add the flex-related mixins to flex-attributes & flex-classes.

**Comment by philmtd** [comment]
Yes, great idea. If you're willing to help "clean up" such inconsistencies in the process then I'll be welcoming these contributions a lot. :)
I think I like creating specific files for the gap properties more than adding them to the other files.

Yes, feel free to move the related mixins to those files. Thank you!

**Comment by philmtd** [comment]
Partly closed by #13
